src/celeritas/io/EventReader.cc

```diff
@@ -115,21 +115,24 @@ auto EventReader::operator()() -> result_type
         // Get the position of the vertex
         auto pos = par->production_vertex()->position();
         HepMC3::Units::convert(pos, evt.length_unit(), HepMC3::Units::CM);
-        primary.position = {pos.x() * units::centimeter,
-                            pos.y() * units::centimeter,
-                            pos.z() * units::centimeter};
+        auto to_cm = [](double v) {
+            return static_cast<real_type>(v) * units::centimeter;
+        };
+        primary.position = {to_cm(pos.x()), to_cm(pos.y()), to_cm(pos.z())};
 
         // Get the lab-frame time [s]
         primary.time = pos.t() * units::centimeter / constants::c_light;
```

Comment:
Should we do this for time as well?
```suggestion
        primary.time = to_cm(pos.t()) / constants::c_light;
```